squire_lib/src/ffi/ffi_tournament.rs

```diff
@@ -637,6 +637,17 @@ impl TournamentId {
         }
     }
 
+    /// Generates a round slip for a tournament
+    /// NULL on error
+    #[no_mangle]
+    pub extern "C" fn tid_round_slips_html(self, __css: *const c_char) -> *const c_char {
```

Comment:
This is still not quite right. Here, you are creating the string `ret`, querying the tournament while not mutating `ret`, and then returning `ret`. Instead, there should not be a `ret`. The query returns an `Option<*const c_char>`. We could handle that like we do in other methods. For example, the tournament name method:
```rust
    pub extern "C" fn tid_name(self) -> *const c_char {
        match SQUIRE_RUNTIME
            .get()
            .unwrap()
            .tournament_query(self, |t| clone_string_to_c_string(&t.name))
        {
            Ok(data) => data,
            Err(err) => {
                print_err(err, "getting tournament name.");
                std::ptr::null()
            }
        }
    }
```